src/compiler/scala/tools/nsc/transform/patmat/Logic.scala

```diff
@@ -356,7 +364,19 @@ trait Logic extends Debugging  {
           // when sym is true, what must hold...
           implied  foreach (impliedSym  => addAxiom(Or(Not(sym), impliedSym)))
           // ... and what must not?
-          excluded foreach (excludedSym => addAxiom(Or(Not(sym), Not(excludedSym))))
+          excluded foreach {
+            excludedSym =>
```

Comment:
Sorry I missed this note! Would it make sense to have the domains stored as sets from the beginning? You're right that `domain(sym) && domain(excludedSym)` is likely faster (especially if we don't have to do `domain.toSet.contains`, though it does mean the closure needs two arguments (`sym`, `excludedSym`, rather than `related`).